src/components/CloudHeader/CloudHeader.js

```diff
@@ -0,0 +1,164 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import ReactDOM from 'react-dom';
+import classNames from 'classnames';
+import CloudHeaderWrapper from './CloudHeaderWrapper';
+import CloudHeaderMenu from './CloudHeaderMenu';
+import CloudHeaderLogo from './CloudHeaderLogo';
+import CloudHeaderList from './CloudHeaderList';
+import CloudHeaderListItem from './CloudHeaderListItem';
+
+export default class CloudHeaderV2 extends React.Component {
+  state = {
+    isMenuOpen: false,
+    isSearchActive: false,
+    isNotificationActive: false,
+    isApplicationActive: false,
+    isUserActive: false,
+  };
+
+  handleMenuClick = () => {
+    this.setState({
+      isMenuOpen: !this.state.isMenuOpen,
+    });
+  };
+
+  handleIconClick = evt => {
+    const type = evt.currentTarget.dataset.icon;
+    const key = `is${type}Active`;
+    this.setState({
+      [key]: !this.state[key],
+    });
+
+    const propFunc = `render${type}`;
+    this.props[propFunc]();
+  };
+
+  componentDidMount() {
+    this.portalNode = document.createElement('div');
+    document.body.appendChild(this.portalNode);
+  }
+
+  componentWillUnmount() {
+    this.portalNode.parentNode.removeChild(this.portalNode);
+  }
+
+  render() {
+    const {
+      className,
+      companyName,
+      productName,
+      logoHref,
+      renderMenu,
+      renderLogo,
+      renderSearch,
+      renderNotification,
+      renderApplication,
+      renderUser,
+      links,
+      ...other
+    } = this.props;
+
+    const searchIcon = (
```

Comment:
Should we move these out of `render` and into the main scope of this module? That way we only pay the cost of initializing this once versus on each render call.

Reply:
Done